lib/Cro/HTTP/Client.pm6

```diff
@@ -156,8 +162,21 @@ class Cro::HTTP::Client {
         }
 
         method send-request($request --> Promise) {
-            my $next-response-promise = Promise.new;
-            $!next-response-vow = $next-response-promise.vow;
+            my $next-response-promise;
+            my $broken = False;
+            $!lock.protect: {
+                if $!dead {
+                    # Without https://github.com/MoarVM/MoarVM/pull/1782 merged,
+                    # we can't put the below return here.
+                    $broken = True;
+                }
+                else {
+                    $next-response-promise = Promise.new;
+                    $!next-response-vow = $next-response-promise.vow;
+                }
+            }
+            return Promise.broken(PipelineClosedBeforeHeaders.new) if $broken;
```

Comment:
Maybe this line would be better in the `if $!dead` arm and no need for var $broken

Reply:
Yes it would be better. But see the comment. ;-)

Reply:
To elaborate a bit more, the PR referenced in the comment actually has been merged. But I don't intentionally want to break Cro for Rakudo versions <= 2024.05 yet.

Reply:
oh - yes I see the comment is related - missed that!